docker/CMSRucioClient/scripts/updateRSEs.py

```diff
@@ -0,0 +1,180 @@
+
+import argparse
+import logging
+
+import requests
+import json
+import re
+
+from functools import wraps
+
+from rucio.client.accountclient import AccountClient
+from rucio.client.rseclient import RSEClient
+from rucio.common.exception import Duplicate, RSEProtocolPriorityError, \
+	RSEProtocolNotSupported, RSENotFound, InvalidObject
+
+DATASVC_URL='http://cmsweb.cern.ch/phedex/datasvc/json/prod/' 
+prog = re.compile('.* -service (.*?) .*')
+
+def setup_logger(logger):
+	""" Code borrowed from bin/rucio-admin
+	"""
+	logger.setLevel(logging.DEBUG)
+	hdlr = logging.StreamHandler()
+	def emit_decorator(fcn):
+		def func(*args):
+			if True:
+				formatter = logging.Formatter("%(message)s")
+			else:
+				levelno = args[0].levelno
+				if levelno >= logging.CRITICAL:
+					color = '\033[31;1m'
+				elif levelno >= logging.ERROR:
+					color = '\033[31;1m'
+				elif levelno >= logging.WARNING:
+					color = '\033[33;1m'
+				elif levelno >= logging.INFO:
+					color = '\033[32;1m'
+				elif levelno >= logging.DEBUG:
+					color = '\033[36;1m'
+				else:
+					color = '\033[0m'
+				formatter = logging.Formatter('{0}%(asctime)s %(levelname)s [%(message)s]\033[0m'.format(color))
+			hdlr.setFormatter(formatter)
+			return fcn(*args)
+		return func
+	hdlr.emit = emit_decorator(hdlr.emit)
+	logger.addHandler(hdlr)
+
+def exception_handler(function):
+	"""Code borrowed from bin/rucio-admin
+	"""
+	@wraps(function)
+	def new_funct(*args, **kwargs):
+		try:
+			return function(*args, **kwargs)
+		except InvalidObject as error:
+			logger.error(error)
+			return error.error_code
+	return new_funct
+
+# Functions for getting PhEDEx information: 
+
+def PhEDEx_node_to_RSE(node, test_tag = '111debug'):
+	""" Translates PhEDEx node names to RSE names. 
+	Make sure new names comply with the policies defined in: 
+	./lib/rucio/common/schema/cms.py
+	./lib/rucio/core/permission/cms.py
+	Because once created RSE name can't be reused, allow to postpend
+	the name wiht a test_tag string
+	"""	
+	return (node+'_'+test_tag).upper()
+
+def PhEDEx_node_FTS_servers(node):
+	"""Returns a list of FTS servers used in node's FileDownload agent configuration"""
+	# FIXME: check node existence. 
+	URL = DATASVC_URL + 'agentlogs?agent=FileDownload&node=' + node
+	RESP = requests.get(url=URL, verify=False)
+	DATA = json.loads(RESP.content)
+	servers = {}
+	for agent in DATA['phedex']['agent']:
+		for log in agent['log']:
+			for message in log['message'].values():
+				assert ('-backend FTS3' in message), \
+					"FTS3 backend is not configured for " + node
+				result = prog.match (message)
+				if result:
+					servers[result.group(1)] = True
+	return servers.keys()
+
+# Functions for translating information to Rucio standards
+
+def PhEDEx_node_names():
+	""" Returns a sorted list of PhEDEx node names via data service nodes API """
+	URL = DATASVC_URL + 'nodes'
+	RESP = requests.get(url=URL, verify=False) # work around cmsweb quirks
```

Comment:
Have a single session object so we don't re-authenticate for each call.  `verify` should work.

See: https://github.com/bbockelm/xrootd-tpc/blob/master/tools/xrootd-test-tpc#L79